aioesphomeapi/connection.py

```diff
@@ -133,8 +147,14 @@ async def _do_cleanup() -> None:
 
                 if self._process_task is not None:
                     self._process_task.cancel()
-                    with suppress(asyncio.CancelledError):
+                    try:
                         await self._process_task
+                    except asyncio.CancelledError:
+                        pass
+                    except Exception as err:  # pylint: disable=broad-except
+                        _LOGGER.error(
+                            "Unexpected exception in process task: %s", err, exc_info=err
+                        )
```

Comment:
So the exception from the process task would bubble here and raise which caused the disconnected callback to never run so we never reconnected